src/transport/MessageCounter.h

```diff
@@ -77,7 +78,8 @@ class GlobalUnencryptedMessageCounter : public MessageCounter
 class LocalSessionMessageCounter : public MessageCounter
 {
 public:
-    static constexpr uint32_t kInitialSyncValue             = 0;          ///< Used for initializing peer counter
+    static constexpr uint32_t kInvalidMessageCounter        = 0;
```

Comment:
why do we use 0 for invalid? Why not kMaxMessageCounter instead?

It seems to me like 0 is a valid value for `generate random 28-bit value`, so using the high bits will cost at most one more value out of roughly 2^32, so not too bad.

Reply:
No, 0 is not a valid value. As per spec:

> All message counters SHALL be initialized with a random value using the `Crypto_DRBG(len = 28) + 1` primitive.

There is a plus 1.